pkg/resource/deployment.go

```diff
@@ -123,9 +142,35 @@ func (gd *generatorDeployment) Create() (runtime.Object, error) {
 }
 
 func (gd *generatorDeployment) Update(o runtime.Object) (runtime.Object, bool, error) {
-	return commonUpdate(gd, o, func(obj runtime.Object) (runtime.Object, error) {
-		return gd.client.Deployments(gd.GetNamespace()).Update(obj.(*appsapi.Deployment))
-	})
+	cur, ok := o.(*appsapi.Deployment)
+	if !ok {
+		return nil, false, fmt.Errorf("invalid runtime.Object: %T", o)
+	}
+
+	n, err := gd.expected()
+	if err != nil {
+		return nil, false, err
+	}
+	exp := n.(*appsapi.Deployment)
+
+	ignore := func(p cmp.Path) bool {
+		last := p.Last().String()
+		return unexportedVar.MatchString(last) || last == ".DefaultMode"
+	}
+
+	if cmp.Equal(exp.Spec, cur.Spec, cmp.FilterPath(ignore, cmp.Ignore())) {
```

Comment:
I'd prefer to check the deployment's Generation instead of guessing what can be defaulted in the deployment.

Like this: https://github.com/openshift/cluster-openshift-apiserver-operator/blob/5cb9b78a3dad4f0b97e0d505eb9522bec8573389///vendor/github.com/openshift/library-go/pkg/operator/resource/resourceapply/apps.go#L88